griptape/utils/observability/griptape_cloud_observability.py

```diff
@@ -0,0 +1,59 @@
+import atexit
+import os
+
+from griptape.utils.observability.griptape_cloud_span_exporter import GriptapeCloudSpanExporter
+from griptape.utils.observability.griptape_instrumentor import GriptapeInstrumentor
+from opentelemetry import trace
+from opentelemetry.trace import Status, StatusCode
+from opentelemetry.instrumentation.threading import ThreadingInstrumentor
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+
+class GriptapeCloudObservability:
+    def __init__(self):
+        self.trace_provider = None
+        self.root_span_context_manager = None
+        self.exit_handler = None
+
+    def start(self):
+        # 1. Configure griptape exporter
+        resource = Resource(attributes={"service.name": "unknown"}, schema_url=None)
+        self.trace_provider = TracerProvider(resource=resource)
+        trace.set_tracer_provider(self.trace_provider)
+        self.trace_provider.add_span_processor(BatchSpanProcessor(GriptapeCloudSpanExporter()))
+
+        # 2. Instrument library
+        GriptapeInstrumentor().instrument(tracer_provider=self.trace_provider)
+        ThreadingInstrumentor().instrument()
+
+        # 3. Create a root span, named "main"
+        tracer = trace.get_tracer("griptape", tracer_provider=self.trace_provider)
+        self.root_span_context_manager = tracer.start_as_current_span("main")  # pyright: ignore
```

Comment:
Why the pyright ignore? If necessary, be more specific i.e. https://github.com/griptape-ai/griptape/blob/888d50ecd5cab443fade30b8786c97d73ecf2e7a/griptape/tools/google_drive/tool.py#L236

Reply:
Made the pyright error more specific. It seems like the libraries isn't providing the correct type hints because pyright is complaining about not calling `start_as_current_span` with the right args, but they are right.